flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroup.java

```diff
@@ -168,21 +168,40 @@ public String getMetricIdentifier(String metricName) {
 	 * @param filter character filter which is applied to the scope components if not null.
 	 * @return fully qualified metric name
 	 */
-	public String getMetricIdentifier(String metricName, CharacterFilter filter) {
-		if (scopeString == null) {
-			if (filter != null) {
-				scopeString = ScopeFormat.concat(filter, registry.getDelimiter(), scopeComponents);
-			} else {
-				scopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
-			}
+	public String getMetricIdentifier(String metricName,CharacterFilter filter) {
+		if (filter != null){
+			scopeString = ScopeFormat.concat(filter, registry.getDelimiter(), scopeComponents);
+		} else {
+			scopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
 		}
-
-		if (filter != null) {
+		if (filter != null){
 			return scopeString + registry.getDelimiter() + filter.filterCharacters(metricName);
 		} else {
 			return scopeString + registry.getDelimiter() + metricName;
 		}
 	}
+
+	/**
+	 * Returns the fully qualified metric name for reporter with certain number position in group, for example
```

Comment:
the javadoc should say "Returns the fully qualified metric name using the configured delimiter for the reporter with the given index.